src/main/java/org/apache/maven/plugins/clean/Cleaner.java

```diff
@@ -115,9 +124,63 @@ public void delete( File basedir, Selector selector, boolean followSymlinks, boo
 
         File file = followSymlinks ? basedir : basedir.getCanonicalFile();
 
+        if ( selector == null && !followSymlinks && fastFolder != null )
+        {
+            if ( fastDelete( file ) )
+            {
+                return;
+            }
+        }
+
         delete( file, "", selector, followSymlinks, failOnError, retryOnError );
     }
 
+    private boolean fastDelete( File baseDir )
+    {
+        fastFolder.mkdirs();
+        if ( fastFolder.isDirectory() )
+        {
+            try
+            {
+                File tmpDir = createTempDir( fastFolder );
+                File dstDir = new File( tmpDir, baseDir.getName() );
+                Files.move( baseDir.toPath(), dstDir.toPath(), StandardCopyOption.ATOMIC_MOVE );
```

Comment:
Problematic when the path leads to a directory on another mountpoint. It will likely throw an exception.

Reply:
Please read carefully about the semantics of move: https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#move(java.nio.file.Path,%20java.nio.file.Path,%20java.nio.file.CopyOption...)

Reply:
Yes, that's why the exception is caught if an atomic move can not be performed.  In which case, the usual deletion will be used.

Reply:
Slightly unrelated, but still related: I think it is _sane_ to expect (maybe define as requirement some where) that local repository MUST BE on same mount. Same thing here, I think it is IMHO OKAY to _assume_ that target fol... sorry, directory is on same mount. (more about why's later)